scripts/codeOptimizer/codeParsers/index.js

```diff
@@ -0,0 +1,33 @@
+// Extracts list of files from the directory and subdirectories, parses them and returns comments
+
+import getFiles from "../helper/getFiles.js";
+import getComments from "../helper/getComments.js";
+import extractTODO from "./Todo.js";
+// import extractFIXME from "./Fixme";
+// import extractDP from "./DP";
+
+// Define an array of extractors
+const extractors = [
+  { name: "TODO", extract: extractTODO },
+  // { name: 'FIXME', extract: extractFIXME },
+  // { name: 'DP', extract: extractDP },
+];
+
+const getCommentsFromDirectory = async (directory) => {
+  const files = await getFiles(directory);
```

Comment:
Low Priority

loop through the file directory
    Fetch a single file -> extract comments for that file -> create a task.